feat: Added test infrastructure and initial component tests

amjedidiah opened the pull request:
## Description
Sets up Jest testing infrastructure and initial tests for the Upup component library.

### Changes
- Configured Jest and JSDOM test environment 
- Added tests for:
  - MetaVersion component
  - checkFileType utility
  - s3GeneratePresignedUrl service
- Set up test coverage requirements (1% across all metrics until tests for prior features are added)

### Test Setup
- Frontend: @testing-library/react
- Backend: Jest mocking for AWS services
- Essential mocks for Google and Framer Motion 

### Usage
```bash
pnpm test           # Run tests
pnpm test:coverage  # View coverage
```

amjedidiah committed:
feat(security): implemented pre-signed URL upload approach

BREAKING CHANGE: Replaced direct cloud storage configuration with pre-signed URLs

- Added pre-signed URL endpoint configuration
- Removed cloud storage credentials from frontend
- Deleted cloud storage config files
- Updated upload logic to use pre-signed URLs
- Simplified configuration interface

amjedidiah committed:
fix(upload): resolved CORS issues in pre-signed URL upload

- Added CORS mode to uploadObject fetch request
- Prettier fixes

BREAKING CHANGE: Requires S3 bucket CORS configuration

amjedidiah committed:
fix(app): build and prettier fixes

amjedidiah committed:
chore(app): undoing unnecessary commit changes

amjedidiah committed:
feat(storage): implemented and refined cloud storage SDK integrations

- Added client-side SDK approach for cloud storage providers
- Implemented AWS integration with presigned URLs
- Added credential refresh and SAS token management
- Implemented progress tracking and retry mechanisms
- Fixed type errors and improve error handling
- Removed unused code and align with latest SDK versions

BREAKING CHANGE: Storage configuration requires provider-specific SDK setup

amjedidiah committed:
feat(storage): implemented Azure SDK integrations

- Added client-side SDK approach for Azure provider
- Implemented Azure integration with presigned URLs
- Implemented progress tracking
- Removed unused code and align with latest SDK versions

amjedidiah committed:
chore(app): removed todo from commit

amjedidiah committed:
refactor(app): streamline storageConfig prop values

- Storage config only requires `provider` and `tokenEndpoint` values

amjedidiah committed:
feat(storage): implemented DO SDK integrations

- Added client-side SDK approach for DO provider
- Implemented DO integration with presigned URLs

amjedidiah committed:
feat(app): s3 compliant support

- Unified s3 client-side upload approach for S3 complaint storage providers

amjedidiah committed:
feat(app): merged azure and s3 SDK into one

- preview url for uploaded files

amjedidiah committed:
fix(app): fixed console warning about ETag header

- made ETag header conditional
- removed `previewUrl` from object returned to FE

amjedidiah committed:
feat(app): backend logic

- changed folder structure to accommodate BE logic

amjedidiah committed:
chore(app): build fixes and clean up

amjedidiah committed:
chore(app): better path resolutions and prettier fixes

amjedidiah committed:
feat(app): dual builds

amjedidiah committed:
feat(app): successful merged builds

amjedidiah committed:
fix(app): prettier fixes

amjedidiah committed:
fix(app): review fixes

- undid setting `Content-Type ` header when sending fetch request from `uploadObject`
- improved logic for `handleError`
- improved validation of required configuration
- Enhanced error handling by including `xhr.status` and `xhr.statusText` when reporting error from `uploadWithProgress`
- fixed issue with build failing when temp dist directory exists

amjedidiah committed:
feat(tests): added comprehensive test setup

- Configured Jest environment for React components and JSDOM
- Added setupTests with necessary mocks for Google, Framer Motion
- Implemented detailed test suite for MetaVersion, checkFileType, and s3GeneratePresignedUrl

BSalaeddin reviewed (commented):
### Suggestions additional Test Cases:

Invalid s3ClientConfig: Test how the function behaves when provided with an invalid or incomplete s3ClientConfig.

```
it('should handle invalid s3ClientConfig', async () => {
    const invalidS3Config = { ...mockS3Config, region: '' } // Empty region
    await expect(
        s3GeneratePresignedUrl({
            fileParams: mockFileParams,
            bucketName: mockBucketName,
            s3ClientConfig: invalidS3Config,
            origin: mockOrigin,
            provider: mockProvider,
        })
    ).rejects.toThrow(UploadError)
})
```

Missing Parameters: Ensure the function properly handles cases where required parameters are missing.

```
it('should throw an error when required parameters are missing', async () => {
    await expect(
        s3GeneratePresignedUrl({
            fileParams: mockFileParams,
            bucketName: '',
            s3ClientConfig: mockS3Config,
            origin: mockOrigin,
            provider: mockProvider,
        })
    ).rejects.toThrow(UploadError)
})

```

amjedidiah committed:
test(s3): added comprehensive tests for s3GeneratePresignedUrl and upgraded coverage reporting

feat:
- Removed Codecov in favour of native Jest coverage reporting in CI
- Added GitHub Actions artifact storage for coverage reports
- Added detailed coverage threshold checks in CI pipeline
- Updated `s3GeneratePresignedUrl` test suite per review to test for Invalid s3ClientConfig and Missing Parameters
- Updated jest config to support coverage requirements

amjedidiah committed:
chore(test): setting threshold to 1 pending when others tests will be added

amjedidiah committed:
chore(app): fixed deprecated artifact actions version

amjedidiah committed:
Merge branch 'master' into feat/test-setup

amjedidiah committed:
Merge branch 'master' into feat/test-setup

amjedidiah committed:
refactor(test): better arrangement in tests in central folder
